plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/DeployJob.java

```diff
@@ -216,9 +220,16 @@ private CloudSdk getCloudSdk(Path credentialFile,
         .exitListener(processExitListener)
         .appCommandMetricsEnvironment(CloudToolsInfo.METRICS_NAME)
         .appCommandMetricsEnvironmentVersion(CloudToolsInfo.getToolsVersion())
-        .appCommandOutputFormat("json")
-        .build();
-    return cloudSdk;
+        .appCommandOutputFormat("json");
+
+    IJavaProject javaProject = JavaCore.create(project);
```

Comment:
The JDK/JRE to use seems like another piece of configuration for the DeployJob,  and the DeployCommandHandler is already doing a lot of configuration.  I think it's easier to understand and predict if configuration computation like this is kept in one place.  We may find that the project JDK may not be sufficient for CloudSdk (e.g., may need Java 8).